### PR TITLE
Make quaternion from unit vectors broadcastable

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
This PR extends the behavior of `quaternion_make_from_unit_vectors` to allow non-unit vectors as well as broadcasting. I tripped over this while porting pygfx (it says unit vectors in the name, but I thought it would simply normalize internally if needed), so I took the opportunity to generalize the behavior, make it broadcastable, and promote the unit test to a property-based one that will test more combinations of vectors.

### CODE REVIEW

File: poetry.lock
Comment (Korijn) [inline poetry.lock:1]: can you revert the changes to the lockfile?

Comment (FirefoxMetzger) [reply poetry.lock:1]: there is a new version of black, which I updated to the latest version.

**Review by Korijn — APPROVED** [review]
